src/server/pps/server/monitor.go

```diff
@@ -197,12 +197,10 @@ func (m *ppsMaster) monitorPipeline(ctx context.Context, pipelineInfo *pps.Pipel
 			defer close(ciChan)
 			return backoff.RetryNotify(func() error {
 				pachClient := m.a.env.GetPachClient(ctx)
-				return pachClient.SubscribeCommit(client.NewRepo(pipeline), "",
-					pipelineInfo.SpecCommit.NewProvenance(),
-					"", pfs.CommitState_READY, func(ci *pfs.CommitInfo) error {
-						ciChan <- ci
-						return nil
-					})
+				return pachClient.SubscribeCommit(client.NewRepo(pipeline), "", "", pfs.CommitState_READY, func(ci *pfs.CommitInfo) error {
```

Comment:
Was this changed just to play it safe and go through all of the commits? Also, this seems like it might hang if the context gets canceled while it is blocked on sending to `ciChan`?

Reply:
This was changed because we can no longer subscribe to commits by their provenance.  I think your concerns about this hanging are valid but out-of-scope here - that was an existing problem.

Reply:
Ohh right, forgot that was removed. I'll add a card for not respecting the context.